pkg/doc/did/doc.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package did
+
+import (
+	"encoding/json"
+	"encoding/pem"
+
+	"encoding/hex"
+
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/pkg/errors"
+)
+
+const (
+	jsonldType         = "type"
+	jsonldID           = "id"
+	jsonldServicePoint = "serviceEndpoint"
+	jsonldController   = "controller"
+
+	// various public key encodings
+	jsonldPublicKeyBase58 = "publicKeyBase58"
+	jsonldPublicKeyHex    = "publicKeyHex"
+	jsonldPublicKeyPem    = "publicKeyPem"
+)
+
+// Doc DID Document definition
+type Doc struct {
+	Context        []string
+	ID             string
+	PublicKey      []PublicKey
+	Service        []Service
+	Authentication []VerificationMethod
+	Created        string
+	Updated        string
+	Proof          Proof
+}
+
+// PublicKey DID doc public key
+type PublicKey struct {
+	ID         string
+	Type       string
+	Controller string
+	Value      []byte
+}
+
+// Service DID doc service
+type Service struct {
+	ID              string
+	Type            string
+	ServiceEndpoint string
+	Properties      map[string]interface{}
+}
+
+// VerificationMethod authentication verification method
+type VerificationMethod struct {
+	PublicKey PublicKey
```

Comment:
We should revisit later - should think about the appropriate pattern to access this field.

Reply:
ok